pkg/server/metrics/metrics.go

```diff
@@ -54,16 +59,40 @@ func newServerMetrics() *ServerMetrics {
 		},
 		[]string{},
 	)
+	connections := prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "grpc_connections",
+			Help:      "Number of current grpc connections, partitioned by service method.",
+		},
+		[]string{
+			"service_method",
+		},
+	)
+	
 	prometheus.MustRegister(latencies)
-	return &ServerMetrics{latencies: latencies}
+	prometheus.MustRegister(connections)
+	return &ServerMetrics{latencies: latencies, connections: connections}
 }
 
 // Reset resets the metrics.
 func (a *ServerMetrics) Reset() {
 	a.latencies.Reset()
+	a.connections.Reset()
```

Comment:
Might be missing something but I didn't think one usually reset a gauge.

Reply:
Thanks, fixed.

I don't see any calls to metrics.Metrics.Reset(), is it called implicitly somewhere that I don't realize?

Reply:
Great question. I'm not a prometheus expert but if this is wired in properly I would think prometheus would call it (directly or indirectly) when it grabbed the metrics. Reset is used to stop double counting on counters as they are cumulative results.